src/assisted_installer_controller/assisted_installer_controller_test.go

```diff
@@ -1098,7 +1098,27 @@ var _ = Describe("installer HostRoleMaster role", func() {
 				assistedController.UpdateNodeLabels(context.TODO(), &wg)
 				wg.Wait()
 			})
+			It("Set label only in Ready node", func() {
+				nodeLabels := `{"node.ocs.openshift.io/storage":""}`
+				k8sNodes := GetKubeNodes(kubeNamesIds)
+				for i, cond := range k8sNodes.Items[0].Status.Conditions {
+					if cond.Type == v1.NodeReady {
+						k8sNodes.Items[0].Status.Conditions[i].Status = v1.ConditionFalse
+					}
+				}
+
+				hosts := create3Hosts(models.HostStatusInstalled, models.HostStageDone, nodeLabels)
+				mockbmclient.EXPECT().GetHosts(gomock.Any(), gomock.Any(), []string{models.HostStatusDisabled, models.HostStatusError}).
```

Comment:
`models.HostStatusDisabled, models.HostStatusError` ??
Why use these host status, should we even add labels to hosts with such status?

Reply:
Those are statuses the we skip(ignore) so we should not bring those hosts at all

Reply:
this test doesn't check anything with host statuses in assisted though we call GetHost with those statuses in function

Reply:
If hosts with these status should be skipped why this test is creating nodes with these statuses and check that they get updated???
I understand that what this test is covering isn't that nodes with these statuses get skipped, but why create misleading nodes entity, why not create a test that checks a valid case? 
/hold

Reply:
Please show me where it creates nodes in this statuses?

Reply:
IT's not creating it's calling `GetHosts`

Reply:
https://github.com/openshift/assisted-installer/blob/master/src/assisted_installer_controller/assisted_installer_controller.go#L1033

In order to run test we need to mock all the calls to all the functions inside tested function please see attached link that explains why this mock is expected here